File: C/Nonlinear equations/Newton-Raphson/newton.c
#include<stdio.h>
#include<math.h>

double f(double x) {
    return x*x*x - x*x + 2;
}

double fprime(double x) {
    return 3*x*x - 2*x;
}

void newton(double x0, double error) {
    double x1;
    while(1) {
        x1 = x0 - f(x0)/fprime(x0);
        if(fabs(x1 - x0) < error) {
            break;
        }
        x0 = x1;
    }
    printf("The root is: %lf", x1);
}

int main() {
    double x0 = -1.5, error = 0.0001;
    newton(x0, error);
    return 0;
}

File: Python/Nonlinear equations/Newton-Raphson/newton.py
import math

def f(x):
    return x*x*x - x*x + 2

def fprime(x):
    return 3*x*x - 2*x

def newton(x0, error):
    while True:
        x1 = x0 - f(x0)/fprime(x0)
        if abs(x1 - x0) < error:
            break
        x0 = x1
    print("The root is: ",x1)

x0 = -1.5
error = 0.0001
newton(x0, error)